<launch>

	<!--head used for ped dataset-->
    <!--node pkg="MODT" type="ST_autolabelling_pcl16" name="ST_autolabelling_pcl16" ns="golfcart" output="screen">
		<param name="laser_frame_id" value="golfcart/front_bottom_lidar"/>
		<param name="odom_frame_id" value="golfcart/odom"/>
		<param name="map_frame_id" value = "golfcart/map"/-->
		
	<node pkg="MODT" type="ST_autolabelling_pcl16" name="ST_autolabelling_pcl16" ns="iMiev" output="screen">
		<param name="laser_frame_id" value="iMiev/front_bottom_lidar"/>
		<param name="odom_frame_id" value="iMiev/odom"/>
		<param name="map_frame_id" value = "iMiev/map"/>
		
	<!--head used for vehicle dataset-->
	<!--node pkg="MODT" type="ST_autolabelling_pcl16" name="ST_autolabelling_pcl16" output="screen"-->

		<!--derived labelling-0, auto labelling 1, detection 2-->
		<param name="program_mode" value="2"/>
		<param name="skip_scan_times" value="1"/>
		<param name="downsample_interval" value="1"/>
		<param name="scanNum_perVector" value="2"/>
		
		<!-- type-1 uses my compressed scan method; type-2 use 3d features-->
		<param name="feature_extraction_type" value="1"/>
		
		<!--for my method: see function "get_vector_V4" -->
		<!--for pose-variant features: 16 + 10*n-->
		<!--for pose-invariant features: 16 + 12*n -2 -->
		<param name="pose_variant_features" value ="true"/>
		
		<!--for 3d features method-->
		<!--method 1 for usr: 12 -->
		<!--method 2 for vfh: 308 -->
		<!--method 3 for esf: 640 -->
		<param name="calc_3dfeautres_method" value="3"/>
		<param name="feature_vector_length" value="36"/>

		<param name="map_image_path" value="$(find MODT)/launch/obstacle_layer_modifiedunknown.png"/>
		<param name="DATMO_model_path" value="$(find MODT)/data_and_model/highway21.model"/>
		<param name="DATMO_scale_path" value="$(find MODT)/data_and_model/highway21.range"/>
		
		<!--parameters for autolabelling of plaza pedestrian data -->
		<!--param name="speed_threshold" value="0.0"/>
		<param name="gating_min_size" value="0.1"/>
		<param name="graph_seg_k" value="0.5"/-->
		
		<!--parameters for derived labelling of campus vehicle data -->
		<param name="speed_threshold" value="2.0"/>
		<param name="gating_min_size" value="0.5"/>
		<param name="graph_seg_k" value="3.0"/>
		
		<param name="search_neighbourNum" value="20"/>
		<param name="control_trainingSample_number" value="10"/>
		<param name="dilation_size" value="1"/>
    </node>
    
	<node name="map_server" pkg="map_server" type="map_server" args="$(find MODT)/launch/obstacle_layer.yaml">
		<param name="frame_id" value = "golfcart/map"/>
		<remap from ="map" to = "obstacle_layer"/>
	</node>
	
	<node pkg="rviz" type="rviz" name="rviz" respawn="false" args="-d $(find MODT)/launch/iMiev.vcg" />
	
</launch>

